packages/xl-odt-exporter/src/odt/odtExporter.tsx

```diff
@@ -0,0 +1,235 @@
+  S extends StyleSchema,
+  I extends InlineContentSchema
+> extends Exporter<
+  B,
+  I,
+  S,
+  React.ReactNode,
+  React.ReactNode,
+  Record<string, string>,
+  React.ReactNode
+> {
+  private automaticStyles: Map<string, React.ReactNode> = new Map();
+  private styleCounter = 0;
+
+  public readonly options: ExporterOptions;
+
+  constructor(
+    protected readonly schema: BlockNoteSchema<B, I, S>,
+    mappings: Exporter<
+      NoInfer<B>,
+      NoInfer<I>,
+      NoInfer<S>,
+      React.ReactNode,
+      React.ReactNode,
+      Record<string, string>,
+      React.ReactNode
+    >["mappings"],
+    options?: Partial<ExporterOptions>
+  ) {
+    const defaults = {
+      colors: COLORS_DEFAULT,
+    } satisfies Partial<ExporterOptions>;
+
+    super(schema, mappings, { ...defaults, ...options });
+    this.options = { ...defaults, ...options };
+  }
+
+  protected async loadFonts() {
+    const interFont = await loadFileBuffer(
+      await import("@shared/assets/fonts/inter/Inter_18pt-Regular.ttf")
+    );
+    const geistMonoFont = await loadFileBuffer(
+      await import("@shared/assets/fonts/GeistMono-Regular.ttf")
+    );
+
+    return [
+      {
+        name: "Inter",
+        fileName: "Inter_18pt-Regular.ttf",
+        data: new Blob([interFont], { type: "font/ttf" }),
+      },
+      {
+        name: "Geist Mono",
+        fileName: "GeistMono-Regular.ttf",
+        data: new Blob([geistMonoFont], { type: "font/ttf" }),
+      },
+    ];
+  }
+
+  public transformStyledText(styledText: StyledText<S>): React.ReactNode {
+    const stylesArray = this.mapStyles(styledText.styles);
+    const styles = Object.assign({}, ...stylesArray);
+
+    if (Object.keys(styles).length === 0) {
+      return styledText.text;
+    }
+
+    const styleName = `BN_T${++this.styleCounter}`;
+
+    // Store the complete style element
+    this.automaticStyles.set(
+      styleName,
+      <StyleStyle style:name={styleName} style:family="text">
+        <StyleTextProperties {...styles} />
+      </StyleStyle>
+    );
+
+    return <TextSpan text:style-name={styleName}>{styledText.text}</TextSpan>;
+  }
+
+  public async transformBlocks(
+    blocks: Block<B, I, S>[],
+    nestingLevel = 0
+  ): Promise<React.ReactNode[]> {
+    const ret: React.ReactNode[] = [];
+    let numberedListIndex = 0;
+
+    for (const block of blocks) {
+      if (block.type === "numberedListItem") {
+        numberedListIndex++;
+      } else {
+        numberedListIndex = 0;
+      }
+
+      const children = await this.transformBlocks(
+        block.children,
+        nestingLevel + 1
+      );
+
+      const content = await this.mapBlock(
+        block as any,
+        nestingLevel,
+        numberedListIndex
+      );
+
+      ret.push(content);
+      if (children.length > 0) {
+        ret.push(...children);
+      }
+    }
+
+    return ret;
+  }
+
+  public async toODTDocument(
+    blocks: Block<B, I, S>[],
+    options?: {
+      header?: string;
```

Comment:
I think we want to be able to use more than a string as header / footer. Can you compare with the other exporters?

Reply:
Indeed, other exporters offer more customizability by using underlying libraries.
In this case, I wasn't sure about exposing the XML/JSX syntax to the end-user - it's a bit unusual and might be confusing for some.

We can export certain elements as JSX components and allow users to use them for more customization, but it'll likely still be just a subset of what's possible;

Reply:
Maybe we can accept not just strings, but stringified XML or even `XMLDocument`s?
You can check the lastest changes for this concept - let me know if it's worth refining.

I'd lean towards only accepting stringified XML, since it's the most flexible.